test/functional/p2p_ibd_txrelay.py

```diff
@@ -7,36 +7,64 @@
 from decimal import Decimal
 
 from test_framework.messages import COIN
+from test_framework.mininode import P2PTxInvStore
 from test_framework.test_framework import BitcoinTestFramework
-from test_framework.util import assert_equal
+from test_framework.util import assert_equal, connect_nodes
 
 MAX_FEE_FILTER = Decimal(9170997) / COIN
 NORMAL_FEE_FILTER = Decimal(100) / COIN
+MAX_TIP_AGE = 24 * 60 * 60
 
 
 class P2PIBDTest(BitcoinTestFramework):
     def set_test_params(self):
-        self.setup_clean_chain = True
+        self.setup_clean_chain = False
         self.num_nodes = 2
         self.extra_args = [["-minrelaytxfee={}".format(NORMAL_FEE_FILTER)]] * self.num_nodes
 
     def run_test(self):
-        self.log.info("Check that nodes set minfilter to MAX_MONEY while still in IBD")
-        for node in self.nodes:
-            assert node.getblockchaininfo()['initialblockdownload']
-            for conn_info in node.getpeerinfo():
-                assert_equal(conn_info['minfeefilter'], MAX_FEE_FILTER)
+        # Generate some blocks so that node0 is funded
+        self.nodes[0].generate(nblocks=101)
+        self.sync_blocks(self.nodes)
+
+        # Force node1 into IBD by restarting and fast-forwarding time.
+        # Fast-forwarding the time by > 24 hours causes node1 to consider its tip too old
+        # See the tip age check in IsInitialBlockDownload().
+        self.restart_node(1)
+        tip_time = self.nodes[1].getblockheader(self.nodes[1].getbestblockhash())['time']
+        self.nodes[1].setmocktime(tip_time + MAX_TIP_AGE + 1)
+        connect_nodes(self.nodes[0], 1)
+        connect_nodes(self.nodes[1], 0)
+
+        self.log.info("Check that nodes set minfilter to MAX_MONEY and do not receive tx invs while still in IBD")
+        assert self.nodes[1].getblockchaininfo()['initialblockdownload']
```

Comment:
I'm worried that this might be intermittent. IsInitialBlockDownload is a latch. The first time it returns false causes it to return false on all subsequent calls. Is there a window between restarting the node and fastforwarding time with `setmocktime` where somthing could call `IsInitialBlockDownload()` and cause it to latch to false?

Reply:
Ooooo....what if i setmocktime backwards to mine the blocks originally? Will it restart with current time, and be outdated from the get go?

Reply:
Yes, I think that would work.

Reply:
Gooooooood point. A few ideas I have:
1. Wondering if I can pass in an argument for `start` to get the test node to start up at a particular mocktime?
2. I could... setmocktime backwards to begin with and maybe when it restarts, it'll be outdated from the get go?

Reply:
1. I don't think there's a way to start a bitcoind node with a mocktime. It can only be set by an RPC after the node has started.
2. Yes, I think this would work!

Reply:
> I'm worried that this might be intermittent. IsInitialBlockDownload is a latch.

Yes, I'm seeing the test fail at that line.

Great stuff, will check out your next push asap @gzhao408!

Reply:
Oo hmmmmm these args are only used in `test_framework.add_nodes` to instantiate the `TestNode` objects, and then can't be changed afterward. So I can't actually restart the node with a new `-mocktime`.

Reply:
Not sure 🤔

Reply:
extra_args can be overwritten for start_node(s)